pyphare/pyphare/pharesee/hierarchy.py

```diff
@@ -752,11 +763,15 @@ def plot2d(self, **kwargs):
                 nbrGhosts = pdat.ghosts_nbr
                 x = pdat.x
                 y = pdat.y
-                nx,ny =x.size, y.size
-                data = data.reshape((nx,ny))
-                data = data[nbrGhosts[0]:-nbrGhosts[0], nbrGhosts[1]:-nbrGhosts[1]]
-                x = np.copy(x[nbrGhosts[0]:-nbrGhosts[0]])
-                y = np.copy(y[nbrGhosts[1]:-nbrGhosts[1]])
+
+                # if nbrGhosts is 0, we cannot do array[0,-0]
+                if np.all(nbrGhosts==np.zeros_like(nbrGhosts)):
+                    x = np.copy(x)
+                    y = np.copy(y)
+                else:
+                    data = data[nbrGhosts[0]:-nbrGhosts[0], nbrGhosts[1]:-nbrGhosts[1]]
```

Comment:
maybe something like [this](https://github.com/PHAREHUB/PHARE/pull/591/files#diff-884f2783892817b4ae5e349e6ddb2741883861e14f5526ecf73854f10a6f99d1R307) could make this a little cleaner?

Reply:
or even

```
data = pdat[patch.box]
```

Reply:
ok replaced by 
```python
                    data = data[patch.box]
```